test/integration/Resends.test.js

```diff
@@ -0,0 +1,170 @@
+import { ethers } from 'ethers'
+import uuid from 'uuid/v4'
+
+import StreamrClient from '../../src'
+
+import config from './config'
+
+const { wait, waitForCondition } = require('streamr-test-utils')
+
+const createClient = (opts = {}) => new StreamrClient({
+    url: config.websocketUrl,
+    restUrl: config.restUrl,
+    auth: {
+        privateKey: ethers.Wallet.createRandom().privateKey,
+    },
+    autoConnect: false,
+    autoDisconnect: false,
+    ...opts,
+})
+
+const MAX_MESSAGES = 10
+const TEST_REPEATS = 10
+
+describe('StreamrClient resends', () => {
+    describe('resend', () => {
+        let client
+        let stream
+        let publishedMessages
+
+        beforeEach(async () => {
+            client = createClient()
+            await client.ensureConnected()
+
+            publishedMessages = []
+
+            stream = await client.createStream({
+                name: uuid(),
```

Comment:
tip: doesn't make a lot of difference but consider using [`lodash/uniqueId`](https://lodash.com/docs/4.17.15#uniqueId) instead of `uuid`.

`uniqueId` generates a string using incrementing numbers and an optional custom prefix e.g. `stream1`, `stream2`.

This can help a lot with debugging async behaviour as the incrementing numbers clearly show what order things occurred in so you can more clearly see misorderings or even cases where data is leaking across tests. e.g. `stream10` arrives in a test where the first stream should have been `stream11`.

The downside is that unlike `uuid` it's not guaranteed to be unique across processes, so when Jest runs suites in parallel, you can end up with more than one `stream1`, but I've found confusion due to this to be rare and the benefits of human readable/sortable uniqueness outweighs the possible downsides. I don't think a universally unique name is necessary here, but if you do want more cross-process uniqueness you can do something like:

```JS
const uid = (prefix) => uniqueId(`p${process.pid}-${prefix}`)
```

Reply:
Good point 👍